### ISSUE TITLE
UserLoginRequest.php not found

### ISSUE THREAD

**Issue by MKahmen** [arussell/AlexaNissanLeaf#1]
Hi Aaron,

I am currently getting an error when calling the skill / in CloudWatch:

> Failed to send request UserLoginRequest.php (404: Not Found)

do you experience the same?

Cheers,
Marius

**Comment by arussell** [comment]
Hi Marius, yeah I've seen this for a couple of days since the v3.3.0 update. I think Nissan may have changed some things on their side, breaking the skill. I'm not sure how extensive the changes are yet.

**Comment by arussell** [comment]
The Europe API endpoints have changed, and it seems to be related to the "previous versions will stop working on 2019-07-29" message displayed in the app:

![apichange](https://user-images.githubusercontent.com/52480/62173977-a7186100-b327-11e9-91c9-743596d6b6a7.jpg) 

My guess is that the `gworchest_160803EC` part of the URL is now different, perhaps using a different datecode. It's still connecting to the same hostname but it's bypassing proxy config on Android, and outright failing when proxied on iOS, so I'm having trouble getting mitmproxy to intercept it. It might need mitmproxy running in transparent proxy mode, with the data being grabbed from an older Android device in order to get the proper URLs, assuming SSL pinning isn't happening (although SSL pinning would explain the errors it threw on iOS when other traffic was working through the proxy).

tl;dr: Nissan changed the API endpoints in v3.3.0 and stopped supporting the old ones on 2019-07-29, breaking all third-party API use. Some detective work is required to learn the new API endpoints.

**Comment by arussell** [comment]
Looks like the new API endpoint has been discovered, more info at https://gitlab.com/tobiaswkjeldsen/dartcarwings/issues/15 -- wil implement this fix as soon as I get a chance.

**Comment by arussell** [comment]
Fixed in 470dba97d33f39ec0b5d1f77fb0aa86c3d52826d!

To update your skill, edit your Lambda and replace the content of index.js and leaf.js with the versions on the [master](https://github.com/arussell/AlexaNissanLeaf) branch.
